lib/rock/webapp/tasks/api.rb

```diff
@@ -155,24 +155,22 @@ def get_operation(name_service, name, operation_name)
                         optional :binary, type: Boolean, default: false
                         optional :init, type: Boolean, default: false
                     end
+
                     get ':name_service/:name/ports/:port_name/read' do
-                        
-                        port = get_port(*params.values_at('name_service', 'name', 'port_name', 'init', 'timeout'))
                         #port = port_by_task_and_name(*params.values_at('name_service', 'name', 'port_name'))
-    
-                        if !port.is_reader?
-                            error! "#{port.name} is an input port, cannot read"
-                        end
-                        
                         if Faye::WebSocket.websocket?(env)
-                            port = port.port.to_async.reader(init: true, pull: true)
+                            proxy = Orocos::Async.proxy("#{params[:name_service]}/#{params[:name]}")
+                            asyncport = proxy.port(params[:port_name]);
                             count = params.fetch(:count, Float::INFINITY)
```

Comment:
Is there a way to check the async port for a "reader", respond_to?(:reader) does not work.